Tests are failing on main

### HydroMT version checks

- [x] I have checked that this issue has not already been reported.
- [x] I have checked that this bug exists on the latest version of HydroMT.

### Reproducible Example

Some tests fail after #1070 was implemented. I am not sure how it should have been resolved, but I guess some tests should have been updated also. The corresponding tests running on the main branch after merging #1070 are these: https://github.com/Deltares/hydromt/actions/runs/11071874956. However, when reverting the change locally, the tests still fail so it seems to be a coincidence that this change caused failing tests, might be related to something else?

### Current behaviour

You see there that two tests are failing, these are related to the change in the linked PR I think:
```
=========================== short test summary info ============================
FAILED tests/_validators/test_data_catalog_validation.py::test_valid_catalog_variants - pydantic_core._pydantic_core.ValidationError: 1 validation error for DataCatalogMetaData
  Value error, Current hydromt version 1.0.0.dev0 is not compatible with version specified in catalog >=1.0a,<2 [type=value_error, input_value={'hydromt_version': '>=1.0a,<2', 'roots': ['']}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.9/v/value_error
FAILED tests/_validators/test_data_catalog_validation.py::test_catalog_metadata_validation - pydantic_core._pydantic_core.ValidationError: 1 validation error for DataCatalogMetaData
  Value error, Current hydromt version 1.0.0.dev0 is not compatible with version specified in catalog >=1.0a,<2 [type=value_error, input_value={'hydromt_version': '>=1....'], 'version': '2023.3'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.9/v/value_error
============= 2 failed, 645 passed, 8 skipped in 670.58s (0:11:10) =============
```

### Desired behaviour

passing tests

### Additional context

_No response_

--- a/hydromt/__init__.py
+++ b/hydromt/__init__.py
@@ -1,7 +1,7 @@
 """HydroMT: Automated and reproducible model building and analysis."""
 
 # version number without 'v' at start
-__version__ = "1.0.0.dev0"
+__version__ = "1.0.1.dev0"
 
 # This is only here to suppress the bug described in
 # https://github.com/pydata/xarray/issues/7259

